__tests__/integration/abort_request.test.ts

```diff
@@ -53,13 +52,59 @@ describe('abort request', () => {
       )
     })
 
+    it('should not throw an error when aborted the second time', async () => {
+      const controller = new AbortController()
+      const selectPromise = client.query({
+        query: 'SELECT sleep(3)',
+        format: 'CSV',
+        abort_controller: controller,
+      })
+
+      setTimeout(() => {
+        controller.abort()
+      }, 50)
+
+      await expect(selectPromise).rejects.toEqual(
+        expect.objectContaining({
+          message: expect.stringMatching('The request was aborted'),
+        })
+      )
+
+      controller.abort('foo bar') // no-op, does not throw here
+    })
+
+    it('should abort when request timeout', async () => {
+      await client.close()
+      client = createTestClient({
+        request_timeout: 50,
+      })
+      const controller = new AbortController()
+      const selectPromise = client.query({
+        query: 'SELECT sleep(3)',
+        format: 'CSV',
+        abort_controller: controller,
+      })
+
+      setTimeout(() => {
+        controller.abort('foo bar') // no-op, does not throw here
+      }, 100) // more than request timeout
+
+      await expect(selectPromise).rejects.toEqual(
+        expect.objectContaining({
+          message: expect.stringMatching('Timeout error'),
+        })
+      )
+
+      controller.abort('bar qaz') // no-op, does not throw here
+    })
+
     it('cancels a select query while reading response', async () => {
       const controller = new AbortController()
       const selectPromise = client
         .query({
           query: 'SELECT * from system.numbers',
           format: 'JSONCompactEachRow',
-          abort_signal: controller.signal as AbortSignal,
```

Comment:
Can we keep `abort_signal` as a part of the external API? I feel like asking to provide `abort_controller` is a leaking implementation detail. We can create an abort controller internally and compose it with the external one (if provided). see https://github.com/whatwg/fetch/issues/905